Avatar Block Exceeds Editor Width When 'Link to User Profile' Option is Enabled

Mukulsingh27 opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- Link this PR to its associated issue with an appropriate keyword: Closes, See, Follow up to, etc. -->
Closes #69180 
It fixes the Avatar block width issue in the editor when the "Link to User Profile" option is enabled. Currently, the block extends beyond the editor's maximum width constraints while maintaining correct frontend display.

<!-- In a few words, what is the PR actually doing? -->

## Why?
The Avatar block's wrapper element is not properly constrained within the editor's width boundaries when the link option is enabled. This creates an inconsistent editing experience, though the frontend display remains unaffected. The fix ensures visual consistency between the editor and frontend rendering.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
- Update the editor's HTML markup to fix the issue
- Ref used: Image block

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

1. Open the WordPress block editor
2. Add a new Avatar block to your post/page
3. In the block settings sidebar, enable the `Link to User Profile` option
4. Verify the Avatar block stays within the editor's maximum width
5. Preview the page to confirm the frontend display remains correct

## Screenshots or screencast <!-- if applicable -->

<!-- If you would like to upload screenshots, feel free to use the table below when it is useful to show the difference between before and after the change. -->

|Before|After|
|-|-|
|![Edit-Post-“Avatar-fix”-‹-gutenberg-—-WordPress-02-27-2025_05_09_PM](https://github.com/user-attachments/assets/59508bd0-e91a-4699-b73a-43e2c5e62153)|![Edit-Post-“Avatar-fix”-‹-gutenberg-—-WordPress-02-27-2025_05_01_PM](https://github.com/user-attachments/assets/c8fcc84c-33e8-49e8-901f-cbb5987bd9f6)|

Mukulsingh27 committed:
Update the Avatar's editor HTML markup

Mukulsingh27 committed:
Revert the addition of newline

Mukulsingh27 commented:
@t-hamano 
As per your suggestion [here](https://github.com/WordPress/gutenberg/pull/69181#pullrequestreview-2617857382), I have updated the HTML markup based on the reference from the Image block. It is now looking and working fine.

t-hamano reviewed (commented):
Thanks for the PR!

The change looks very good, but we will need to address three issues this change introduces.

1. The cursor changes to a pointer when the image is linked. Like other linked blocks, the editor should use the default cursor.
2. We can't click on the block "properly". The blue outline doesn't appear when we click on the block, because that would mean we click on the a element.
3. The `a` element is 100% wide. The width of the `a` element should match the width of the image, just like in the frontend.



https://github.com/user-attachments/assets/1a6e084f-e424-4c16-867a-461a68a066b7



For the first and second issues, we will need styles like this in the editor, [just like in the Site Logo block](https://github.com/WordPress/gutenberg/blob/e093fefd041eb6cc4a4e7f67b92ab54fd75c8858/packages/block-library/src/site-logo/editor.scss#L11):

```css
.wp-block-avatar__link {
	pointer-events: none;
}
```

For the third issue, we could use the approach in [this comment](https://github.com/WordPress/gutenberg/issues/69255#issuecomment-2694481520). We will need this style in both the frontend and the editor:

```css
.wp-block-avatar__link {
	display: block;
	justify-self: start;
}
```

Mukulsingh27 committed:
Fix relevant issues introduced by the recent changes

Mukulsingh27 commented:
Thanks @t-hamano for testing and providing the feedback. I’ve updated the PR with the suggested changes to improve the functionality. Could you please confirm that everything looks good?

https://github.com/user-attachments/assets/092baa2a-5494-4b11-bc06-e7f8b8929949

Mukulsingh27 committed:
Move styles from editor to frontend

Mukulsingh27 committed:
Fix the issue with resizable box

Mukulsingh27 commented:
Thanks @t-hamano  for testing and providing the feedback. 

> The cursor changes to a pointer when the image is linked. Like other linked blocks, the editor should use the default cursor.

The cursor is now fixed across the entire width, except for the image. However, if we use `pointer-events: none;`, the resizable box functionality will break because the resize cursors are also wrapped inside the anchor tag.

Therefore, an additional style is added to change the cursor without breaking the resizable box.

> We can't click on the block "properly". The blue outline doesn't appear when we click on the block, because that would mean we click on the a element.

This is now fixed. The blue outline is appearing and working fine.

> The a element is 100% wide. The width of the a element should match the width of the image, just like in the frontend.

If we use these properties:
```CSS
.wp-block-avatar__link {
	display: block;
	justify-self: start;
}
```

then I doubt that the alignxcenter functionality will break. To ensure it works without breaking the functionality, `display: inline-block;` would be more suitable.


https://github.com/user-attachments/assets/739ac83e-fc33-4cdd-a1cf-7cdab841a468

Made necessary changes to fix all the issue.

t-hamano commented:
@Mukulsingh27 Thanks for the update!

> However, if we use `pointer-events: none;`, the resizable box functionality will break because the resize cursors are also wrapped inside the anchor tag. Therefore, an additional style is added to fix the cursor without breaking the resizable box.

This approach is fine, but it is not recommended to rely on selectors for specific components. What do you think about changing the following configuration, like for the Site Logo block?

```html
<div class="block-editor-block-list__block wp-block wp-block-avatar">
  <div class="components-resizable-box__container>
  <a href="#avatar-pseudo-link" class="wp-block-avatar__link">
      <img class="wp-block-avatar__image">
    </a>
  </div>
</div>
```

> > The a element is 100% wide. The width of the a element should match the width of the image, just like in the frontend.
> 
> If we use these properties:
> ```CSS
> .wp-block-avatar__link {
> 	display: block;
> 	justify-self: start;
> }
> ```
> 
> Then I doubt that the `aligncenter` functionality will break. To ensure it works without breaking the functionality, `display: inline-block;` would be more suitable.

Sorry, this approach was certainly not appropriate. The problems with this approach were recently discussed here: 
 https://github.com/WordPress/gutenberg/issues/69255#issuecomment-2695424214

Mukulsingh27 committed:
Update the markup and remove the CSS dependency

Mukulsingh27 commented:
Thanks @t-hamano for sharing the configuration.
I've taken the reference from the site-logo block and updated the avatar block to follow the same structure in terms of markup, removing the CSS dependency. Here is the final version. Can you please confirm? Thanks.


https://github.com/user-attachments/assets/f938b5ed-45f4-4e7f-800a-bc49c420ac5c
